src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2661,13 +2661,21 @@ private BoundExpression BindCatchFilter(CatchFilterClauseSyntax filter, Diagnost
             BoundExpression boundFilter = this.BindBooleanExpression(filter.FilterExpression, diagnostics);
             if (boundFilter.ConstantValue != ConstantValue.NotAvailable)
             {
-                Error(diagnostics, ErrorCode.WRN_FilterIsConstant, filter.FilterExpression);
+                // Depending on whether the filter constant is true or false, and whether there are other catch clauses,
+                // we suggest different actions
+                var errorCode = boundFilter.ConstantValue.BooleanValue
+                    ? ErrorCode.WRN_FilterIsConstantTrue
+                    : ((filter.Parent.Parent as TryStatementSyntax).Catches.Count == 1) && ((filter.Parent.Parent as TryStatementSyntax).Finally == null)
```

Comment:
(filter.Parent.Parent is TryStatementSyntax s && s.Catches.Count == 1 && s.Finally == null)